nomos-services/storage/src/backends/surrealdb.rs

```diff
@@ -0,0 +1,171 @@
+use crate::backends::{StorageBackend, StorageSerde, StorageTransaction};
+use async_trait::async_trait;
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+use std::marker::PhantomData;
+use std::str::from_utf8;
+use surrealdb::engine::any::{connect, Any};
+use surrealdb::Surreal;
+use tokio::runtime::Handle;
+use tokio::task::block_in_place;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Data {
+    pub data: Bytes,
+}
+
+impl Data {
+    fn new(value: Bytes) -> Self {
+        Self { data: value }
+    }
+
+    fn to_bytes(&self) -> Bytes {
+        self.data.clone()
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Surreal(#[from] surrealdb::Error),
+    #[error(transparent)]
+    Utf8(#[from] std::str::Utf8Error),
+}
+
+#[derive(Clone)]
+pub struct SurrealBackendSettings {
+    pub db_path: String,
+    pub namespace: String,
+    pub database: String,
+}
+
+impl SurrealBackendSettings {
+    pub fn new(db_path: String, namespace: String, database: String) -> Self {
+        Self {
+            db_path,
+            namespace,
+            database,
+        }
+    }
+}
+
+pub struct SurrealBackendTransaction {}
+
+impl StorageTransaction for SurrealBackendTransaction {
+    type Result = Result<Option<Bytes>, Error>;
+    type Transaction = Self;
+}
+
+pub struct SurrealBackend<SerdeOp>
+where
+    SerdeOp: StorageSerde + Send + Sync + 'static,
+{
+    db: Surreal<Any>,
+    _serde_op: PhantomData<SerdeOp>,
+}
+
+#[async_trait]
+impl<SerdeOp> StorageBackend for SurrealBackend<SerdeOp>
+where
+    SerdeOp: StorageSerde + Send + Sync + 'static,
+{
+    type Settings = SurrealBackendSettings;
+    type Error = Error;
+    type Transaction = SurrealBackendTransaction;
+    type SerdeOperator = SerdeOp;
+
+    fn new(config: Self::Settings) -> Result<Self, Self::Error> {
+        let Self::Settings {
+            db_path,
+            namespace,
+            database,
+        } = config;
+
+        let connect_to_surrealdb = async {
+            let db = connect(db_path).await?;
+            db.use_ns(namespace).use_db(database).await?;
+            Ok(Self {
+                db,
+                _serde_op: Default::default(),
+            })
+        };
+
+        block_in_place(|| Handle::current().block_on(connect_to_surrealdb))
+    }
+
+    async fn store(&mut self, key: Bytes, value: Bytes) -> Result<(), Self::Error> {
+        let key = from_utf8(&*key)?;
+        let data = Data::new(value);
+        let _record: Option<Data> = self.db.create(("kv", key)).content(data).await?;
+        Ok(())
+    }
+
+    async fn load(&mut self, key: &[u8]) -> Result<Option<Bytes>, Self::Error> {
+        let key = from_utf8(key)?;
+        let record: Option<Data> = self.db.select(("kv", key)).await?;
+        Ok(record.map(|data| data.to_bytes()))
```

Comment:
Weird that `clippy` does not throw a warning for this:
 ```suggestion
        Ok(record.map(Data::to_bytes))
```